src/NTDll/NTDll.cs

```diff
@@ -42,6 +42,31 @@ public static partial class NTDll
             Kernel32.ACCESS_MASK desiredAccess,
             [Friendly(FriendlyFlags.In)] OBJECT_ATTRIBUTES* objectAttributes);
 
+        /// <summary>
+        /// The RtlVerifyVersionInfo routine compares a specified set of operating system version requirements to the
+        /// corresponding attributes of the currently running version of the operating system.
+        /// </summary>
+        /// <param name="VersionInfo">Pointer to an <see cref="Kernel32.OSVERSIONINFOEX"/> structure that specifies the
+        /// operating system version requirements to compare to the corresponding attributes of the currently running
+        /// version of the operating system.</param>
+        /// <param name="TypeMask">Specifies which members of VersionInfo to compare with the corresponding attributes of
+        /// the currently running version of the operating system. </param>
+        /// <param name="ConditionMask">Specifies how to compare each VersionInfo member. To set the value of ConditionMask,
+        /// a caller should use the <see cref="Kernel32.VerSetConditionMask(long, Kernel32.VER_MASK, Kernel32.VER_CONDITION)"/> function.</param>
+        /// <returns>
+        ///     <see cref="NTSTATUS.Code.STATUS_SUCCESS"/> when the specified version matches the currently running version of the operating system.
+        ///     <see cref="NTSTATUS.Code.STATUS_INVALID_PARAMETER"/> when the input parameters are not valid.
+        ///     <see cref="NTSTATUS.Code.STATUS_REVISION_MISMATCH"/> when the specified version does not match the currently running version of the operating system.
+        /// </returns>
+        /// <remarks>
+        ///     See remarks in <see cref="Kernel32.VerifyVersionInfo(ref Kernel32.OSVERSIONINFOEX, Kernel32.VER_MASK, long)"/>.
+        ///
+        ///     Unmanifested applications that call <see cref="RtlVerifyVersionInfo(ref Kernel32.OSVERSIONINFOEX, Kernel32.VER_MASK, long)"/> are not
+        ///     suspectible to version-lies by the OS.
+        /// </remarks>
+        [DllImport(nameof(NTDll))]
+        public static extern NTSTATUS RtlVerifyVersionInfo(ref Kernel32.OSVERSIONINFOEX VersionInfo, Kernel32.VER_MASK TypeMask, long ConditionMask);
```

Comment:
We should use a native pointer for `VersionInfo` instead of `ref`. Then use the `[Friendly(FriendlyFlags.Bidirectional)]` attribute so that more friendly overloads like your `ref` here are automatically generated.

Example:

https://github.com/AArnott/pinvoke/blob/a561cd6ca12e2190db01c89e5a43ba7b3e1f71a2/src/Kernel32/storebanned/Kernel32.cs#L858